plugins/header_rewrite/operators.h

```diff
@@ -109,16 +109,17 @@ class OperatorSetRedirect : public Operator
 public:
   OperatorSetRedirect() { TSDebug(PLUGIN_NAME_DBG, "Calling CTOR for OperatorSetRedirect"); }
   void initialize(Parser &p) override;
+
   TSHttpStatus
-  get_status()
+  get_status() const
   {
     return static_cast<TSHttpStatus>(_status.get_int_value());
   }
-  std::string
-  get_location(int &size)
+
+  const std::string &
```

Comment:
I'd be careful here. This returns a reference to something that could be a temporary, depending on what type `get_value()` returns. If it's really a reference to memory owned by `_location`, I'd return `std::string_view` to make that clear. If you can't return that, you can't return a `std::string` reference either.

Reply:
I'm not changing to string_view in any of this code.

Reply:
Also, the std::string is a member of the Value, and get_value() already returns a reference.